legacy/reqcounter/reqcounter.go

```diff
@@ -24,28 +24,71 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// RequestCounter records the number of HTTP requests to GCR.
-type RequestCounter struct {
-	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
-	requests uint64     // Number of HTTP requests since recording started.
-	since    time.Time  // When the current counter began recording requests.
+// requestCounter records the number of HTTP requests to GCR.
+// TODO: @tylerferrara in the future, add a field 'persistent bool' to determine if
+// the counter should ever reset.
+type requestCounter struct {
+	mutex    sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
+	requests uint64        // Number of HTTP requests since recording started.
+	since    time.Time     // When the current counter began recording requests.
+	interval time.Duration // The duration of time between each log.
 }
 
 // log records the number of HTTP requests found and resets the counter.
-func (rc *RequestCounter) log() {
+func (rc *requestCounter) log() {
```

Comment:
nit: consider naming this as `flush()` (which is used typically for such use cases)

Reply:
Sure thing!